Source/DafnyCore/DooFile.cs

```diff
@@ -236,8 +241,10 @@ public class ManifestData {
     }
   }
 
-  public static void RegisterNoChecksNeeded(params Option[] options) {
-    foreach (var option in options) {
+  public static void RegisterNoChecksNeeded(Option option, bool semantic) {
```

Comment:
Since we also have `TranslationRecord.RegisterLibraryChecks`, I think we could move option classification somewhere more central.

It becomes tempting to add more metadata to the `Option`s directly. That might work, but my one persistent requirement is that every time a new option is added, we should firmly encourage explicitly thinking about what kind of option it is, and not have any default classification that might be picked up without thinking about it.

Reply:
Can we keep translation record options out of scope for this PR?

Reply:
Yup I realized this after commenting too, that's reasonable since this is only an internal API, we can centralize it later. I was also thinking it would be more efficient to audit all of the options options for both characteristics at once, but seems like you've finished with scanning them for the `semantic` flag we can scan for translation relevance later too.